src/bmconfigparser.py

```diff
@@ -150,68 +157,40 @@ def items(self, section, raw=False, variables=None):
-
-    if sys.version_info[0] == 3:
-        @staticmethod
-        def addresses(hidden=False):
-            """Return a list of local bitmessage addresses (from section labels)"""
-            return [x for x in BMConfigParser().sections() if x.startswith('BM-') and (
-                    hidden or not BMConfigParser().safeGetBoolean(x, 'hidden'))]
-
-        def read(self, filenames):
-            SafeConfigParser.read(self, filenames)
-            for section in self.sections():
-                for option in self.options(section):
-                    try:
-                        if not self.validate(
-                                section, option,
-                                self[section][option]
-                        ):
-                            try:
-                                newVal = BMConfigDefaults[section][option]
-                            except configparser.NoSectionError:
-                                continue
-                            except KeyError:
-                                continue
-                            SafeConfigParser.set(
-                                self, section, option, newVal)
-                    except configparser.InterpolationError:
-                        continue
-
-    else:
-        @staticmethod
-        def addresses():
-            """Return a list of local bitmessage addresses (from section labels)"""
-            return [
-                x for x in BMConfigParser().sections() if x.startswith('BM-')]
-
-        def _reset(self):
-            """Reset current config. There doesn't appear to be a built in
-               method for this"""
-            sections = self.sections()
-            for x in sections:
-                self.remove_section(x)
-
-        def read(self, filenames):
-            """Read config and populate defaults"""
-            self._reset()
-            SafeConfigParser.read(self, filenames)
-            for section in self.sections():
-                for option in self.options(section):
-                    try:
-                        if not self.validate(
-                                section, option,
-                                SafeConfigParser.get(self, section, option)
-                        ):
-                            try:
-                                newVal = BMConfigDefaults[section][option]
-                            except KeyError:
-                                continue
-                            SafeConfigParser.set(
-                                self, section, option, newVal)
-                    except configparser.InterpolationError:
-                        continue
+        return ConfigParser.items(self, section, True, variables)
+
+    @staticmethod
+    def addresses(hidden=False):
+        """Return a list of local bitmessage addresses (from section labels)"""
+        return [x for x in BMConfigParser().sections() if x.startswith('BM-') and (
```

Comment:
maybe this can work with `self.sections()` instead of `BMConfigParser().sections()`? Then you can keep it as a method.